src/enemy.hpp

```diff
@@ -18,9 +19,10 @@ class Enemy {
 
 	void render(le::Renderer& renderer) const;
 	void translate(kvf::Seconds dt);
-	// There are temporary parameters, will take the light beam object once it is created
-	void check_collision(glm::vec2 pos, float radius);
+	void take_damage(std::size_t dmg);
 	[[nodiscard]] std::size_t get_health() const { return m_health; }
+	[[nodiscard]] std::pair<glm::vec2, float> get_pos_diameter() const;
```

Comment:
Don't use `std::pair`, create a struct, then it and its members will have meaningful names rather than just `.first`.